app/collector/data_collection_diff_checker.py

```diff
@@ -0,0 +1,102 @@
+
+import logging
+from datetime import datetime
+import json
+
+class DataCollectionDiffChecker():
+# Class to implement the collection data engine.
+
+    def __init__(self, log):
+        self.log = log
+
+
+    def _diff_list (self, k, new_dict, current_dict, obj_type):
+    # Method to do a diff check between two list objects.
+        diff = False
+
+        # Iterate over the new values and search for their match in the old values.
+        for new in new_dict[k]:
+            match = False
+            for current in current_dict[k]:
+                
+                # Find the elements with the same name.
+                if new['name'] == current['name']:
+                    match = True
+                    for n in new.keys():
+                        
+                        # Check if there was any difference.
+                        if (new[n] != current[n]):
+                            diff = True
+                            # Log the difference found.
+                            self.log.log_diff_list_element (obj_type, current_dict["id"], current_dict["fully_qualified_name"], k, n, current[n], new[n])
```

Comment:
Long line. Set your IDE to display a vertical line at column 81. Break line if it exceeds the limit.